vm/src/stdlib/os.rs

```diff
@@ -2290,6 +2291,46 @@ mod nt {
         get_stats().map_err(|e| convert_io_error(vm, e))
     }
 
+    // cwait is available on MSVC only (according to CPython)
+    #[cfg(target_env = "msvc")]
+    extern "C" {
+        fn _cwait(termstat: *mut i32, procHandle: intptr_t, action: i32) -> intptr_t;
+        fn _get_errno(pValue: *mut i32) -> i32;
+    }
+
+    #[cfg(target_env = "msvc")]
+    #[pyfunction]
+    fn waitpid(pid: intptr_t, opt: i32, vm: &VirtualMachine) -> PyResult<(intptr_t, i32)> {
+        const ECHILD: i32 = 10;
+        const EINVAL: i32 = 22;
+
+        let mut status = 0;
+        let pid = unsafe { suppress_iph!(_cwait(&mut status, pid, opt)) };
+        if pid == -1 {
+            let mut errno = 0;
+            unsafe { _get_errno(&mut errno) };
+            match errno {
+                ECHILD => Err(vm.new_exception_msg(
+                    vm.ctx.exceptions.os_error.clone(),
+                    "ECHILD: No spawned processes".to_owned(),
+                )),
+                EINVAL => Err(vm.new_exception_msg(
+                    vm.ctx.exceptions.os_error.clone(),
+                    "EINVAL: Invalid argument".to_owned(),
+                )),
```

Comment:
Following unix `convert_nix_error`, here could be `convert_nt_error`

Reply:
I'd love to use that kind of error handler, but it seems like in `_cwait` Microsoft uses some kind of POSIX-like errno which means it's neither WinAPI error nor POSIX nix error.